dev/com.ibm.ws.cdi.cts_fat_tck/fat/src/com/ibm/ws/cdi/cts/tck/CdiCtsTckLauncher.java

```diff
@@ -0,0 +1,134 @@
+ */
+@RunWith(FATRunner.class)
+public class CdiCtsTckLauncher {
+
+    /**  */
+    private static String GIT_REPO_PARENT_DIR = "publish/gitRepos/";
+    private static final String CDI_TCK_REPO_NAME = "cdi-tck";
+    // Weld Porting Package
+    private static final String WAS_CTS_REPO_NAME = "liberty-cdi-tck-runner";
+
+    @Server("CdiCtsTCKServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        server.startServer();
+    }
+
+    /**
+     * Various TCK tests test for Deployment, Definition and other Exceptions and
+     * these will cause the test suite to be marked as FAILED if found in the logs
+     * when the server is shut down. So we tell Simplicity to allow for the message
+     * ID's below.
+     *
+     * @throws Exception
+     */
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+    }
+
+    /**
+     * Run the TCK (controlled by autoFVT/publish/tckRunner/tcl/tck-suite.html)
+     *
+     * @throws Exception
+     */
+    @Mode(TestMode.EXPERIMENTAL)
+    @Test
+    @AllowedFFDC // The tested exceptions cause FFDC so we have to allow for this.
+    public void runFreshMasterBranchTck() throws Exception {
+
+        File repoParent = new File(GIT_REPO_PARENT_DIR);
+
+        //Build and install locally the actual latest master branch CDI TCK
+        File repo = new File(repoParent, CDI_TCK_REPO_NAME);
+        MvnUtils.mvnCleanInstall(repo);
+
+        HashMap<String, String> addedProps = new HashMap<String, String>();
+
+        // Fetch the API version to inject into the runner pom.xml via a property
+        String apiVersion = MvnUtils.getApiVersionAfterClone(repo);
+        System.out.println("Queried api.version is : " + apiVersion);
+        addedProps.put("api.version", apiVersion);
+
+        // Fetch the TCK version to inject into the runner pom.xml via a property
+        String tckVersion = MvnUtils.getTckVersionAfterClone(repo);
+        System.out.println("Queried tck.version is : " + tckVersion);
+        addedProps.put("tck.version", tckVersion);
+
+        // A command line -Dprop=value is transformed to a ENV variable
+        // by the Simplicity framework
+        //
+        // Allow the impl version to be overridden
+        String implVersion = System.getenv("impl.version");
+        System.out.println("Passed in impl.version is : " + implVersion);
+        addedProps.put("impl.version", implVersion);
+
+        // See other *TckLauncher classes for examples of how to use the below:
+        HashSet<String> versionedLibraries = null;
+        String backStopImpl = null;
+
+        MvnUtils.runTCKMvnCmdWithProps(server, "com.ibm.ws.cdi.cts_fat_tck", this.getClass() + ":launchCdiCtsTck",
+                                       addedProps, versionedLibraries, backStopImpl);
+    }
+
+    @Mode(TestMode.LITE)
+    @Test
+    public void testThatDoesNothingAndCanAlwaysRunAndPass() {
```

Comment:
We already have a libraray method for this. Have a look at com.ibm.ws.microprofile.reactive.streams.operators_fat_tck/fat/src/com/ibm/ws/microprofile/reactive/streams/operators/tck/FATSuite.java for an example.